programs/bpf_loader/src/syscalls.rs

```diff
@@ -2515,18 +2520,22 @@ mod tests {
         let addr = &pubkey as *const _ as u64;
         let config = Config::default();
         let memory_mapping = MemoryMapping::new::<UserError>(
-            vec![MemoryRegion {
-                host_addr: addr,
-                vm_addr: 100,
-                len: std::mem::size_of::<Pubkey>() as u64,
-                vm_gap_shift: 63,
-                is_writable: false,
-            }],
+            vec![
+                MemoryRegion::default(),
```

Comment:
What is driving these changes to the tests?

Reply:
The reason I ask is because the the memory mapping `map` call changed which could be a change in behavior.  Looks like the vm address's must now be in the known ranges that the vm uses and can no longer be arbitrary values.  This looks fine since the VM doesn't use these arbitrary vm addresss, but want to make sure.

Reply:
Yes, the layout of `MemoryRegion`s in the `MemoryMapping` is now stricter so the tests had to be adjusted. We did not take the memory layout from the ELF and instead enforced a static one. So I don't think anything changed for the programs and they are still running on my validator node.